internal/handler/handler_test.go

```diff
@@ -0,0 +1,746 @@
+	reader := strings.NewReader(string(jsonFirstEmployee))
+	testTable := []struct {
+		URL      string
+		method   string
+		expected int
+		read     io.Reader
+	}{
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 200,
+			read:     reader,
+		},
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 500,
+			read:     strings.NewReader("s"),
+		},
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 400,
+			read:     strings.NewReader(string(jsonThirdEmployee)),
+		},
+		{
+			URL:      "http://localhost:8080/employee",
+			method:   "POST",
+			expected: 400,
+			read:     strings.NewReader(string(jsonSecondEmployee)),
+		},
+	}
+	for _, testCase := range testTable {
+		r, err := http.NewRequest("POST", "http://localhost:8080/employee", testCase.read)
+		r.Header.Set("content-type", "application/json")
+		r = createTestContext(r)
+		if err != nil {
+			t.Fatalf("could not created request: %v", err)
+		}
+		w := httptest.NewRecorder()
+		handler.CreateEmployee(w, r)
+		result := w.Result()
+		defer result.Body.Close()
+		if result.StatusCode != testCase.expected {
+			t.Fatalf("expected status created; get %v", result.StatusCode)
+		}
+	}
+}
+
+func TestHand_CreatePosition(t *testing.T) {
+	initTest()
+	firstPosition := internal.Position{Salary: decimal.New(500, 0), Name: "worker"}
+	fakePosition := internal.Employee{ID: createEmpID(), FirstName: "V", LasName: "T", PositionID: uuid.New()}
+	jsonFirstPosition, _ := json.Marshal(firstPosition)
+	jsonFakePosition, _ := json.Marshal(fakePosition)
+	reader := strings.NewReader(string(jsonFirstPosition))
+	testTable := []struct {
+		URL      string
+		method   string
+		expected int
+		read     io.Reader
+	}{
+		{
+			URL:      "http://localhost:8080/position",
+			method:   "POST",
+			expected: 500,
+			read:     strings.NewReader("s"),
+		},
+		{
+			URL:      "http://localhost:8080/position",
+			method:   "POST",
+			expected: 200,
+			read:     reader,
+		},
+		{
+			URL:      "http://localhost:8080/position",
+			method:   "POST",
+			expected: 400,
+			read:     strings.NewReader(string(jsonFakePosition)),
+		},
+		{
+			URL:      "http://localhost:8080/position",
+			method:   "POST",
+			expected: 400,
+			read:     strings.NewReader(string(jsonFirstPosition)),
+		},
+	}
+	for _, testCase := range testTable {
+		r, err := http.NewRequest("POST", "http://localhost:8080/position", testCase.read)
+		r.Header.Set("content-type", "application/json")
+		r = createTestContext(r)
+		if err != nil {
+			t.Fatalf("could not created request: %v", err)
+		}
+		w := httptest.NewRecorder()
+		handler.CreatePosition(w, r)
+		result := w.Result()
+		defer result.Body.Close()
+		if result.StatusCode != testCase.expected {
+			t.Fatalf("expected status created; get %v", result.StatusCode)
+		}
+	}
+}
+
+func TestHand_DeleteEmployeeVarsZero(t *testing.T) {
```

Comment:
That test doesn't check anything useful related to business logic.

Reply:
Yes, but I add this case for covering error which is related with length of mux.Vars(...). I shouldn't do it for covering?

Reply:
+